Ports Barometers from Yogstation

KomradeSpectre opened the pull request:
:cl: Super3222, TheMythicGhost, DaedalusGame
add: Adds a barometer function to the standard atmos analyzer.
imageadd: Adds a new sprite for the atmos analyzer to resemble a barometer.
/:cl:

[why]: Despite the unpredictability of storms being a 'FEATURE', having some way of predicting future storms can assist miners to make decisions to tough out the storm, seek shelter, or return to base more intuitively than waiting for the storm warning and making a mad dash. This also can be extended to work with future weather types by adding barometer_predictable = TRUE to said weather type.

Port from https://github.com/yogstation13/yogstation/pull/1285 with some shoehorning.

KomradeSpectre committed:
Ports Barometers from Yogstation

KomradeSpectre committed:
Ports Barometers from Yogstation

KomradeSpectre committed:
Added missing barometer_predictable bools

DaedalusGame commented:
Sprites are from Oracle (uraniummeltdown pr'd them before, but they got denied for gas meters)

Repukan commented:
Miners get a good 10 second warning to seek shelter, they don't need a way to power game storms or it removes the only danger it had (unpredictability)

WJohn commented:
No. We have analyzers that handle this already. Throw this function into those so they're slightly more useful and so these things aren't a one-trick object that will never be taken.

KomradeSpectre commented:
If you can lead me to what item you're meaning that has the weather detecting functionality, by all means. I ported this because it was incredibly useful and immersive in Yog, and I thought it would be here.

WJohn commented:
The air analyzer found in all mechanical toolboxes. The black and blue handheld scanner that everyone *immediately* throws the fuck away because it's effectively worthless.

DaedalusGame commented:
it's worthless on lavaland because it scans atmosphere and all of lavaland has the same atmosphere

Qustinnus commented:
what are you on about @DaedalusGame

DaedalusGame commented:
you'll figure it out eventually, try reading the comment again

DaedalusGame commented:
but they don't look like barometers

ChuckTheSheep commented:
If you're going to be pedantic, why would spessmen use analogue barometers? Don't miners have an non-atmos analyzer already? (I think it might be for ore) if you care about the oldtimey look Just slap a pressure meter on that thing.

optimumtact commented:
wjohn already neatly answered why this is not a great change.

I know it's more fun to add unique sprites, but existing stuff needs a lot lot more love

KomradeSpectre committed:
Gives the atmos analyzer love instead of a new item.

KomradeSpectre commented:
I've changed the sprite for the atmos analyzer and removed my old barometer sprite. We'll have to wait for the IconDiff bot to regenerate the preview.

Cyberboss commented:
It's broken atm, post it yourself please

KomradeSpectre commented:
The different icon is as follows:
![image](https://user-images.githubusercontent.com/12734451/36439378-b629c35e-163a-11e8-9dee-6f0c7c8f21ed.png)

nicbn commented:
Can you make it a different item instead of a mode please?

DaedalusGame commented:
leave

KomradeSpectre commented:
>Can you make it a different item instead of a mode please?

**AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA!!!!!!!**

nicbn commented:
Analyzer already is pretty useful on its own for atmos techs. Miners should get a different tool.

DaedalusGame commented:
go argue with oranges and wjohn

nicbn commented:
No you go

DaedalusGame commented:
i already did further up

ChuckTheSheep commented:
Tie it into the ore scanner...

DaedalusGame commented:
>shoehorn it into a different device

ChuckTheSheep commented:
Still related to mining.

KomradeSpectre commented:
I honestly feel this should be reviewed again as I have completed what was originally requested, and it works as intended.

DaedalusGame commented:
Chuck, this also works in snowdin

ChuckTheSheep commented:
:man_shrugging:

optimumtact commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -403,6 +406,69 @@ GAS ANALYZER
+		if(!T)
+			return
+
+		playsound(src, 'sound/effects/pop.ogg', 100)
```
shouldn't you put this after the cooldown?

KomradeSpectre committed:
Moves playsound under cooldown to prevent soundspam.

KomradeSpectre commented:
Done as requested @optimumtact
